packages/backend-proxy-middleware/src/base/proxy.ts

```diff
@@ -0,0 +1,292 @@
+        if (destination) {
+            authNeeded = destination.Authentication === 'NoAuthentication';
+            // in case of a full url destination remove the path defined in the destination from the forwarded call
+            if (isFullUrlDestination(destination)) {
+                const destPath = new URL(destination.Host).pathname.replace(/\/$/, '');
+                if (backend.path.startsWith(destPath) && !backend.pathPrefix) {
+                    backend.pathPrefix = backend.path.replace(destPath, '');
+                }
+            }
+        } else {
+            throw new Error();
+        }
+    }
+    return authNeeded;
+}
+
+/**
+ * Enhance the proxy options with information read from the store.
+ *
+ * @param proxyOptions reference to a proxy options object that the function will enhance
+ * @param system backend system information (most likely) read from the store
+ * @param oAuthRequired if true then the OAuth flow is triggered to get cookies
+ * @param tokenChangedCallback function to call if a new refreshToken is available
+ */
+export async function enhanceConfigForSystem(
+    proxyOptions: Options & { headers: object },
+    system: BackendSystem,
+    oAuthRequired: boolean | undefined,
+    tokenChangedCallback: (refreshToken?: string) => void
+): Promise<void> {
+    if (oAuthRequired) {
+        if (system.serviceKeys) {
+            const provider = createForAbapOnBtp(
+                JSON.parse(system.serviceKeys as string),
+                system.refreshToken,
+                tokenChangedCallback
+            );
+            // sending a request to the backend to get cookies
+            await provider.getAtoInfo();
+            proxyOptions.headers!['cookie'] = provider.cookies.toString();
+        } else {
+            // TODO: similar to prompting user/password, should allow prompting for service keys here?
+            throw new Error('Cannot connect to ABAP Environment on BTP without service keys.');
+        }
+    } else if (system.authenticationType === AuthenticationType.ReentranceTicket) {
+        // TODO: @ullas needs to add the missing code to the axios-extension
+        //const connection = await (await system.getCatalog(ODataVersion.v2)).getConnection();
+        //cookies = connection.cookies;
+    } else {
+        if (
+            (system.username || process.env.FIORI_TOOLS_USER) &&
+            (system.password || process.env.FIORI_TOOLS_PASSWORD)
+        ) {
+            proxyOptions.auth = `${system.username || process.env.FIORI_TOOLS_USER}:${
+                system.password || process.env.FIORI_TOOLS_PASSWORD
+            }`;
+        }
+    }
+}
+
+export async function getBackendProxy(
+    backend: BackendConfig,
+    common: CommonConfig,
+    logger: Logger
+): Promise<RequestHandler> {
+    await initI18n();
+    // base options
+    const proxyOptions: Options & { headers: object } = {
+        secure: !common.ignoreCertError,
```

Comment:
Can we rename `ignoreCertError` to `secure`? I already did it for the `ui5-proxy-middleware`. 
`ignoreCertError` is a property of the old proxy that we used in the beginning.

Reply:
https://github.com/SAP/open-ux-tools/pull/471/commits/37da9e071daf8571cb401e1c462f9716b9601acd